src/main/java/com/sekwah/advancedportals/bukkit/listeners/Listeners.java

```diff
@@ -62,45 +61,29 @@ public static void reloadValues(AdvancedPortalsPlugin plugin) {
         WandMaterial = Material.getMaterial(ItemID);
     }
 
-    @EventHandler
-    public void onJoinEvent(PlayerJoinEvent event) {
-        // TODO: make relability
-        // Portal.cooldown.put(event.getPlayer().getName(), System.currentTimeMillis());
-
-        /* if (plugin.PlayerDestiMap.containsKey(event.getPlayer())) {
-            String desti = plugin.PlayerDestiMap.get(event.getPlayer());
-
-            Destination.warp(event.getPlayer(), desti);
-        } */
-    }
-
-    @EventHandler
-    public void onWorldChangeEvent(PlayerChangedWorldEvent event) {
-        // TODO: make relability
-        // Portal.cooldown.put(event.getPlayer().getName(), System.currentTimeMillis());
-    }
-
-    @EventHandler(ignoreCancelled = true)
-    public void onTeleportEvent(PlayerTeleportEvent event) {
-        // TODO: make relability
-        // Portal.cooldown.put(event.getPlayer().getName(), System.currentTimeMillis());
-    }
-
     @EventHandler(ignoreCancelled = true)
     public void spawnMobEvent(CreatureSpawnEvent event) {
-        if(event.getSpawnReason() == CreatureSpawnEvent.SpawnReason.NETHER_PORTAL && Portal.inPortalRegion(event.getLocation(), Portal.getPortalProtectionRadius())) {
+        if (event.getSpawnReason() == CreatureSpawnEvent.SpawnReason.NETHER_PORTAL
+                && Portal.inPortalRegion(event.getLocation(), Portal.getPortalProtectionRadius())) {
             event.setCancelled(true);
         }
     }
 
-    @EventHandler(ignoreCancelled = true)
-    public void onLeaveEvent(PlayerQuitEvent event) {
```

Comment:
If onLeave is being removed please add a scheduled task that removes data for users that no longer have cooldowns. Otherwise on more popular server there will be more and more data being used by the plugin permanently. Yes its very little data but it is not good practice to leave data that may never be touched again unattended. Even if it only runs once a day.

Reply:
Since HashMap doesn't shrink itself when data is removed from it, there is no actual difference (especially in performance) between master and this pr, I'm not sure if such optimization should be related with this pr.